### DISCUSSION TITLE
Allow inspection and/or manipulation of HTTP headers?

### DISCUSSION THREAD

**Discussion by lkrms** [r0x0r/pywebview#1201] [Ideas]
First, thank you for creating pywebview. From what I've seen so far, it's a bit of a gem, deserving of wider success :)

I'm using pywebview as a WebKit2-GTK fallback for a macOS fork of this [SAML VPN authentication helper](https://github.com/lkrms/gp-saml-gui). I'd like to open an upstream PR where WebKit2-GTK is replaced entirely, but at the moment, this would involve loss of functionality with endpoints that only return SAML response data in HTTP headers (most endpoints copy these to the body of the response as an HTML comment--which pywebview allows me to extract--but not all).

So far, I haven't been able to find any reference to HTTP headers in pywebview's documentation or examples, but if I've missed a way of surfacing them via a `webview` instance, I look forward to being educated 😄 

Failing that, is an API for this on the radar for pywebview? It appears to be possible with WKWebView and Edge/Chromium backends, but I haven't gone too deep just yet. I don't have a lot of available time but am interested in picking away at this if it would fit with your vision for pywebview, especially if you have a preferred API in mind.

**Comment by r0x0r** [comment]
Handling http headers has been requested several times, but no work has been on this front.
Header support would be nice indeed. I am not sure about API, but maybe something like this would work

Request headers:
`window.load_url(url: string, headers: Dict)`

Response headers:
The `window.events.loading`  event with the handler signature `on_loading(response_headers: Dict)`
